ledger-core/v2/virtual/execute/execute.go

```diff
@@ -441,6 +475,8 @@ func (s *SMExecute) setNewState(prototype reference.Global, memory []byte) func(
 			memory,
 			parentReference,
 		))
+
+		state.DecrementPotentialPendingCounter(!s.execution.Unordered)
```

Comment:
Bugs here:
- setNewState action on shared data is not always called, for example it is not called when there is no new state produced, this should be covered with a unit test on stepSaveNewObject
- counters of potential pendings shouldn't be changed after pulse change, they are not potential anymore. As we don't use other measures to finalize object state, implemented behaviour will result in different VStateReports on multiple VStateRequests.

Reply:
1. Don't change counters after pulse chafe
2. fix problem in case of there is no new state